cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -1401,262 +1252,184 @@ unbundleProtocolParams (BundleAsShelleyBasedProtocolParameters _ pp _) = pp
+      protocolParamDecentralization
+    , protocolParamExtraPraosEntropy
+    , protocolParamMinUTxOValue
+    } = do
+  ppCommon <- toShelleyCommonPParams protocolParameters
+  d <- case protocolParamDecentralization of
+    Nothing -> Left "Missing Decentralization parameter"
+    Just dr -> boundRationalEither "D" dr
+  minUTxOValue <-
+    maybe (Left "toShelleyPParams: must specify protocolParamMinUTxOValue") Right
+    protocolParamMinUTxOValue
+  let ppShelley =
+        ppCommon
+        & ppDL            .~ d
+        & ppExtraEntropyL .~ toLedgerNonce protocolParamExtraPraosEntropy
+        & ppMinUTxOValueL .~ toShelleyLovelace minUTxOValue
+  pure ppShelley
+
+
+toAlonzoCommonPParams :: AlonzoEraPParams ledgerera
+                      => ProtocolParameters
+                      -> Either String (PParams ledgerera)
+toAlonzoCommonPParams
+    protocolParameters@ProtocolParameters {
+      protocolParamCostModels
+    , protocolParamPrices
+    , protocolParamMaxTxExUnits
+    , protocolParamMaxBlockExUnits
+    , protocolParamMaxValueSize
+    , protocolParamCollateralPercent
+    , protocolParamMaxCollateralInputs
+    } = do
+  ppShelleyCommon <- toShelleyCommonPParams protocolParameters
+  costModels <- toAlonzoCostModels protocolParamCostModels
+  prices <-
+    requireParam "protocolParamPrices" toAlonzoPrices protocolParamPrices
+  maxTxExUnits <-
+    requireParam "protocolParamMaxTxExUnits" Right protocolParamMaxTxExUnits
+  maxBlockExUnits <-
+    requireParam "protocolParamMaxBlockExUnits" Right protocolParamMaxBlockExUnits
+  maxValueSize <-
+    requireParam "protocolParamMaxBlockExUnits" Right  protocolParamMaxValueSize
+  collateralPercent <-
+    requireParam "protocolParamCollateralPercent" Right protocolParamCollateralPercent
+  maxCollateralInputs <-
+    requireParam "protocolParamMaxCollateralInputs" Right protocolParamMaxCollateralInputs
+  let ppAlonzoCommon =
+        ppShelleyCommon
+        & ppCostModelsL           .~ costModels
+        & ppPricesL               .~ prices
+        & ppMaxTxExUnitsL         .~ toAlonzoExUnits maxTxExUnits
+        & ppMaxBlockExUnitsL      .~ toAlonzoExUnits maxBlockExUnits
+        & ppMaxValSizeL           .~ maxValueSize
+        & ppCollateralPercentageL .~ collateralPercent
+        & ppMaxCollateralInputsL  .~ maxCollateralInputs
+  pure ppAlonzoCommon
+
+toAlonzoPParams :: Ledger.Crypto crypto
+                => ProtocolParameters
+                -> Either String (PParams (Ledger.AlonzoEra crypto))
+toAlonzoPParams
+    protocolParameters@ProtocolParameters {
+      protocolParamDecentralization
+    , protocolParamUTxOCostPerWord
+    } = do
+  ppAlonzoCommon <- toAlonzoCommonPParams protocolParameters
+  -- QUESTION? This is strange, why do we need to construct Alonzo Tx with Babbage PParams?
+  -- This feels to me like an issue with the api design, as there should never be such an
+  -- incosistency, because PParams affect the validity of the transaction.
```

Comment:
`s/incosistency/inconsistency`

Who is this question for?

Reply:
To whoever wrote this comment. Which seems like @Jimbo4350 in this commit: bbb8a5702eeef990462befdba85d52cc79602744
```
         -- The decentralization parameter is deprecated in Babbage
         -- so we default to 0 if no decentralization parameter is found
         -- in the api's 'ProtocolParameter' type. If we don't do this
         -- we won't be able to construct an Alonzo tx using the Babbage
         -- era's protocol parameter because our only other option is to
         -- error.
```

Reply:
Does that resolve your question @lehins ?

Reply:
It sure does. Thank you. Removed the hacky implementation in favor of the correct one

Reply:
Turns out removing the hack and making `d` parameter required breaks some tests. I've reverted the hack back and we can tackle it at a later PR

Reply:
Turned this comment into an issue: https://github.com/input-output-hk/cardano-node/issues/5093